lib/events-reader-checkpoint-writer.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+const EventEmitter = require('events');
+const debug = require('debug')('events-reader');
+let writerLoopStopped = true;
+let lastDoc;
+let lastCheckpointId;
+let harvesterApp;
+
+const checkpointEvent = new EventEmitter();
+
+checkpointEvent.on('newCheckpoint', (checkpointId, doc) => {
+  lastCheckpointId = checkpointId;
+  lastDoc = doc;
+});
+
+const persistLastCheckpoint = () => {
+  if(lastDoc && lastCheckpointId) {
+    harvesterApp.adapter.update('checkpoint', lastCheckpointId, {ts: lastDoc.ts})
+      .then(checkpoint => {
+        debug('last written checking point ' + checkpoint.ts);
+      })
+      .catch(error => {
+        console.log(error);
+        process.exit(1);
+      });
+  };
+  lastCheckpointId = undefined;
+  lastDoc = undefined;
+};
+
+const persistInInterval = (ms) => {
+  setInterval(() => {
```

Comment:
you should probably save this in a global variable (like writerLoopIsRunning) so you can tell that the interval is running, and so you can call clearInterval (if you want to) to stop it. Since stop isnt required right now, you dont have to (and shouldnt) write that method now.